src/cli/net/server/mod.rs

```diff
@@ -21,23 +26,69 @@ pub mod handlers;
 pub enum MpcServerError {
     #[error(transparent)]
     BadQueryString(#[from] QueryRejection),
+    #[error("header not found: {0}")]
+    MissingHeader(String),
+    #[error("invalid header: {0}")]
+    InvalidHeader(BoxError),
+    #[error(transparent)]
+    BadPathString(#[from] PathRejection),
+    #[error(transparent)]
+    HttpError(#[from] hyper::Error),
+    #[error("parse error: {0}")]
+    SerdeError(#[from] serde_json::Error),
+    #[error("could not forward messages: {0}")]
+    SendError(BoxError),
+}
+
+/// [`From`] implementation for [`MpcServerError::SendError`]
+impl<T: Debug + Send + Sync + 'static> From<mpsc::error::SendError<T>> for MpcServerError {
```

Comment:
nit: it may be better to drop these requirements for `T` as `SendError` is literally just "channel closed" error.  You could implement this for any `T` if you convert this string to `Error` (there is a From implementation for that)

Reply:
probably for the best, because this is also driving `Step` to be `Sync`, since `T` here must be `Sync`